Formula/imgur-screenshot.rb

```diff
@@ -1,18 +1,30 @@
 class ImgurScreenshot < Formula
   desc "Take screenshot selection, upload to imgur. + more cool things"
   homepage "https://github.com/jomo/imgur-screenshot"
-  url "https://github.com/jomo/imgur-screenshot/archive/v1.7.4.tar.gz"
-  sha256 "1f0f2d5e201f1fdc1472f201f04430d809bf442ad034c194e70d8921823e990e"
+  url "https://github.com/jomo/imgur-screenshot/archive/v2.0.0.tar.gz"
+  sha256 "1581b3d71e9d6c022362c461aa78ea123b60b519996ed068e25a4ccf5a3409f5"
   license "MIT"
   head "https://github.com/jomo/imgur-screenshot.git"
 
   bottle :unneeded
 
+  depends_on "jq"
+  depends_on "terminal-notifier"
+
+  uses_from_macos "curl"
+
+  on_macos do
+    depends_on "bash"
```

Comment:
Why is bash (4) needed? This line seems to indicate older versions are supported: https://github.com/jomo/imgur-screenshot/blob/68635afa1aefa423a626e0d010517bdfd42bdb41/imgur-screenshot#L76

Reply:
No idea. My guess is that line is a leftover from the previous version which did fully support older versions of bash.

Reply:
`declare -g` was only introduced in bash 4.2. Since that's used all over the script, it can be argued that the test you noticed is redundant.

Reply:
Yes, I guess the version test is broken. It even uses `declare -g` in the `else` statement.

Reply:
On a different note: I just noticed that all core formulae that `depends_on "bash"` don't restrict it to macOS-only. Perhaps we can dispense with the `on_macos` qualifier?

Reply:
I'd rather not. I actually suspect that it's all those other formulae are the ones that need changing.

Reply:
We don't introduce linux-specific things or `mac_only` unless the formula is tested on Linux CI, which this one is not. So let's depend on bash unconditionally.

Reply:
Agreed. We try to keep dependencies between mac and linux as close as possible.
Differences should only be there if 1) something does not build on a specific platform 2) it is not needed on a specific platform 3) we have really good reasons to have a difference

Reply:
I was thinking this falls into case 2, actually.